src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -499,6 +500,23 @@ void ov::npuw::CompiledModel::finalize_weights_bank() {
     LOG_INFO("Done.");
 }
 
+void ov::npuw::CompiledModel::detach_memory() {
+    LOG_INFO("Detaching model & weight memory...");
+    LOG_BLOCK();
+    for (size_t idx = 0; idx < m_compiled_submodels.size(); ++idx) {
+        auto& comp_model_desc = m_compiled_submodels[idx];
```

Comment:
Maybe, as an alternative to calling `.detach()` in the bank itself, it could've been called here for lazy tensors in the closure - @smirnov-alexey consider this change for the future

Reply:
I had it initially but it required additional checks. I prefer this implementation since it's desired behavior for every case anyway

Reply:
However in case of having some CPU-compiled subgraphs I'd implement something like this https://github.com/openvinotoolkit/openvino/pull/27415/files#diff-2ed7db3ac71f25124dadd21bb7a90ec92b2ba192cbf9a864e83828ba77ef6926R565 - we need to copy CPU consts to and reset the ones from the original model

Reply:
Can be a FIXME for the future